### PR TITLE
Added gql integration docs

### LINKED ISSUE #7944: GraphQL Docs

### PR DESCRIPTION
**Pull request by szokeasaurusrex**
Closes #7944

## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [x] Checked Vercel preview for correctness, including links
- [x] PR was reviewed and approved by any necessary SMEs
- [x] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

*Describe your changes here. If your PR relates to or resolves an issue, add a link to that too.*

## Extra resources

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

**Review by sentrivana — APPROVED** [review]

File: src/platforms/python/common/configuration/integrations/gql.mdx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+---
+title: gql
+description: gql GraphQL client integration
```
Comment (shanamatthews) [inline src/platforms/python/common/configuration/integrations/gql.mdx:3]:
```suggestion
title: GQL
description: GQL GraphQL client integration
```

File: src/platforms/python/common/configuration/integrations/gql.mdx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+sidebar_order: 55
+---
+
+The [gql](https://gql.readthedocs.io/en/stable/) integration allows you to monitor all GraphQL requests that your Python application makes using the `gql` library. Specifically, this integration reports every [`TransportQueryError`](https://gql.readthedocs.io/en/latest/modules/transport_exceptions.html#gql.transport.exceptions.TransportQueryError) to Sentry and provides additional context including the URL the request was made to, GraphQL query which caused the error, and the `errors` information that the GraphQL server returns.
```
Comment (shanamatthews) [inline src/platforms/python/common/configuration/integrations/gql.mdx:7]:
```suggestion
The [GQL](https://gql.readthedocs.io/en/stable/) integration allows you to monitor all GraphQL requests that your Python application makes using the `gql` library. Specifically, this integration reports every [`TransportQueryError`](https://gql.readthedocs.io/en/latest/modules/transport_exceptions.html#gql.transport.exceptions.TransportQueryError) to Sentry and provides additional context including:
- The URL the request was made to
- The GraphQL query which caused the error
- The `errors` information that the GraphQL server returns
```

Comment (shanamatthews) [reply src/platforms/python/common/configuration/integrations/gql.mdx:7]: suggest capitalization fix and breaking up features into a bulleted list for readability

File: src/platforms/python/common/configuration/integrations/gql.mdx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        GQLIntegration(),
+    ],
+    # Uncomment the following line to send GraphQL
+    # queries and error information to Sentry:
+    #
+    # send_default_pii=True,
+)
+```
+
+### Data Privacy & The `gql` Integration
```
Comment (shanamatthews) [inline src/platforms/python/common/configuration/integrations/gql.mdx:39]: This seems like it should be its own H2 heading

```suggestion
## Data Privacy and the GQL Integration
```

File: src/platforms/python/common/configuration/integrations/gql.mdx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+To ensure data privacy, the [SDK avoids sending any information which may potentially contain personally identifiable information (PII)](/platforms/python/configuration/options/#send-default-pii) by default.
```
Comment (shanamatthews) [inline src/platforms/python/common/configuration/integrations/gql.mdx:41]:
```suggestion
To ensure data privacy, the SDK avoids sending any information that may potentially contain personally identifiable information (PII) by default. See the docs for the [`send_default_pii` flag](/platforms/python/configuration/options/#send-default-pii) for more information.
```

File: src/platforms/python/common/configuration/integrations/gql.mdx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+For the `gql` integration, this policy means that the SDK **refrains from sending the GraphQL queries and the `errors` information** that the GraphQL server returns unless you override the default configuration via the `send_default_pii` flag.
```
Comment (shanamatthews) [inline src/platforms/python/common/configuration/integrations/gql.mdx:43]:
```suggestion
For the GQL integration, this policy means that the SDK **doesn't send** the GraphQL queries and the `errors` information that the GraphQL server returns, unless you override the default configuration via the `send_default_pii` flag.
```

File: src/platforms/python/common/configuration/integrations/gql.mdx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+<Alert level="warning">
+  Prior to enabling the <code>send_default_pii</code> flag, please ensure that
+  you have configured Sentry's{" "}
+  <a href="/platforms/python/data-management/sensitive-data/">
+    sensitive data scrubbing
+  </a>{" "}
+  tools properly to scrub all PII.
```
Comment (shanamatthews) [inline src/platforms/python/common/configuration/integrations/gql.mdx:51]: I believe you can use normal markdown if you put leading & trailing newlines in the alert

```suggestion

Before enabling the `send_default_pii` flag, ensure you've configured Sentry's [sensitive data scrubbing](/platforms/python/data-management/sensitive-data/) tools to scrub all PII.
  
```

File: src/platforms/python/common/configuration/integrations/gql.mdx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+</Alert>
+
+If you would like Sentry to record the GraphQL queries and the `errors` information returned by the GraphQL server, you can uncomment the `send_default_pii=True` line in the code snippet above.
```
Comment (shanamatthews) [inline src/platforms/python/common/configuration/integrations/gql.mdx:54]:
```suggestion
To have Sentry record the GraphQL queries and the `errors` information returned by the GraphQL server, uncomment the `send_default_pii=True` line in the code snippet above.
```

**Review by shanamatthews — APPROVED** [review]
Looks good! Just some small suggestions. I recommended changing gql -> GQL based on their preferred capitalization in their docs.

**Pull request merged**